### PR TITLE
BUGFIX: CONFIG_THERMO not works

### PR DESCRIPTION
**Pull request by husxk**
The GNU standard says:
"Identifiers that are not macros, which are all considered to be the number zero. This allows you to write #if MACRO instead of #ifdef MACRO, if you know that MACRO, when defined, will always have a nonzero value. Function-like macros used without their function call parentheses are also treated as zero."

According to that, pt100 and ktype were identifiers that are not macros, and was considered as 0.
So we were always including, no matter what the user has passed.

### CODE REVIEW

**Comment by husxk** [comment]
fix for #16

File: Makefile [hunk 1]
```diff
@@ -4,12 +4,29 @@ GREEN := $(ESC)[0;32m
...
 CONFIG_THERMO := Ktype
 CONFIG_MAGNETRON  := 0
 CONFIG_HOSTNAME   := "pico_furnace"
 CONFIG_WATER := 1
+CONFIG_THERMO_INTERNAL = $(CONFIG_THERMO_NONE)
+
+ifeq ($(CONFIG_THERMO),Ktype)
+	CONFIG_THERMO_INTERNAL=$(CONFIG_THERMO_KTYPE)
+else ifeq ($(CONFIG_THERMO),pt100)
+	CONFIG_THERMO_INTERNAL=$(CONFIG_THERMO_PT100)
```
Comment (SecMeant) [inline Makefile:21]: Let's error out when user pass invalid value

**Review by SecMeant — CHANGES_REQUESTED** [review]
Looks good, but let's report an error in case user misspell a config value

File: max318xx.h [hunk 2]
```diff
@@ -56,11 +56,10 @@ static inline uint8_t max318xx_write_reg8(uint8_t addr, uint8_t val)
...
 
-
-#if CONFIG_THERMO == pt100
+#if CONFIG_THERMO == CONFIG_THERMO_PT100
   #include "max31865.h"
-#elif COFNIG_THERMO == Ktype
+#elif CONFIG_THERMO == CONFIG_THERMO_KTYPE
   #include "max31856.h"
-#else
+#elif CONFIG_THERMO >= 1
```
Comment (SecMeant) [inline max318xx.h:63]: `CONFIG_THERMO > CONFIG_THERMO_MAX`?

**Review by SecMeant — CHANGES_REQUESTED** [review]

**Comment by SecMeant** [comment]
Applied the changes locally and now testing. I'll merge this manually. Thanks!

**Pull request closed without merging**